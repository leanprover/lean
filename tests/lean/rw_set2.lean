open nat

constant f : nat → nat
constant g : nat → nat

axiom foo : ∀ x, x > 0 → f x = 0 ∧ g x = 1
axiom bla : ∀ x, g x = f x + 1

attribute foo [simp]
attribute bla [simp]

set_option pp.notation false
print [simp] simp
