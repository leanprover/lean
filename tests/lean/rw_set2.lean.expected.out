simplification rules for iff
#1, ?M_1 ↔ ?M_1 ↦ true
#1, false ↔ ?M_1 ↦ ¬?M_1
#1, ?M_1 ↔ false ↦ ¬?M_1
#1, true ↔ ?M_1 ↦ ?M_1
#1, ?M_1 ↔ true ↦ ?M_1
#0, false ↔ true ↦ false
#0, true ↔ false ↦ false
#1, ¬?M_1 ↔ ?M_1 ↦ false
#1, ?M_1 ↔ ¬?M_1 ↦ false
#2, ?M_1 - ?M_2 ≤ ?M_1 ↦ true
#1, 0 ≤ ?M_1 ↦ true
#1, succ ?M_1 ≤ ?M_1 ↦ false
#1, pred ?M_1 ≤ ?M_1 ↦ true
#1, ?M_1 ≤ succ ?M_1 ↦ true
#1, ?M_1 ∧ ?M_1 ↦ ?M_1
#1, ?M_1 ∧ ¬?M_1 ↦ false
#1, ¬?M_1 ∧ ?M_1 ↦ false
#1, false ∧ ?M_1 ↦ false
#1, ?M_1 ∧ false ↦ false
#1, true ∧ ?M_1 ↦ ?M_1
#1, ?M_1 ∧ true ↦ ?M_1
#3 perm, ?M_1 ∧ ?M_2 ∧ ?M_3 ↦ ?M_2 ∧ ?M_1 ∧ ?M_3
#3, (?M_1 ∧ ?M_2) ∧ ?M_3 ↦ ?M_1 ∧ ?M_2 ∧ ?M_3
#2 perm, ?M_1 ∧ ?M_2 ↦ ?M_2 ∧ ?M_1
#2, ?M_2 == ?M_2 ↦ true
#2, ?M_1 - ?M_2 < succ ?M_1 ↦ true
#1, ?M_1 < 0 ↦ false
#1, ?M_1 < succ ?M_1 ↦ true
#1, 0 < succ ?M_1 ↦ true
#1, ?M_1 ∨ ?M_1 ↦ ?M_1
#1, false ∨ ?M_1 ↦ ?M_1
#1, ?M_1 ∨ false ↦ ?M_1
#1, true ∨ ?M_1 ↦ true
#1, ?M_1 ∨ true ↦ true
#3 perm, ?M_1 ∨ ?M_2 ∨ ?M_3 ↦ ?M_2 ∨ ?M_1 ∨ ?M_3
#3, (?M_1 ∨ ?M_2) ∨ ?M_3 ↦ ?M_1 ∨ ?M_2 ∨ ?M_3
#2 perm, ?M_1 ∨ ?M_2 ↦ ?M_2 ∨ ?M_1
#2, ?M_2 = ?M_2 ↦ true
#2, ¬?M_2 = ?M_2 ↦ false
#0, ¬false ↦ true
#0, ¬true ↦ false
simplification rules for eq
#1, g ?M_1 ↦ f ?M_1 + 1
#2, g ?M_1 ↦ 1
#2, f ?M_1 ↦ 0
#3, ite false ?M_2 ?M_3 ↦ ?M_3
#3, ite true ?M_2 ?M_3 ↦ ?M_2
#4, ite ?M_1 ?M_4 ?M_4 ↦ ?M_4
#1, 0 - ?M_1 ↦ 0
#2, succ ?M_1 - succ ?M_2 ↦ ?M_1 - ?M_2
